en/debug/profiling.md

```diff
@@ -12,6 +12,28 @@ The folded stacks are then fed into the visualization script, for which purpose
 
 ## Basic Usage
 
+### Prerequisites
+
+The profiler relies on GDB to run PX4 on the embedded target.
+So before profiling a target, you must have the hardware you wish to profile, and you must compile and upload the firmware to that hardware.
+You will then need a JTAG debugger, such as the DroneCode Probe, to run the GDB server and interact with the board.
+You may also need to modify the contents of `poor-mans-profiler.sh` to point to the debugger.
+For example, with a Pixhawk 4 connected via USB, followed by a DroneCode Probe over USB, the following devices are enumerated in Ubuntu:
```

Comment:
What does this mean "with a Pixhawk 4 connected via USB, followed by a DroneCode Probe over USB," - I think maybe "followed" is incorrect, because it implies a daily chain - Pixhawk 4 > USB > Dronecode probe, while you probably mean Pixhawk 4 debug port > Dronecode probe > Usb > laptop  (do you?). Maybe an image?

My thinking is that this section changes now you have removed hard coding for the port right? I think you can say that if you're using the Dronecode probe or black magic probe you can just accept the default. Otherwise then you can use the command (like you've shown below) to locate the port to use. 

Can you also link to http://docs.px4.io/master/en/debug/swd_debug.html#debug-probes because I'm not sure that everyone knows about jtag probes and how to connect them to PX4.

Reply:
PS This is definitely needed, so thanks for taking it on.

Reply:
Good point.  I'll update the wording based on the changes I just made.